### PR TITLE
feat: resource provider card for admin dashboard

### PR DESCRIPTION
**Pull request by debjyoti-pandit**
UI looks like this - 

image

when none of the resources are there - 

image

overall

image

### CODE REVIEW

File: packages/components/dashboard/resource-provider-card/resource-provider-card.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    ).reduce((p, [k, v]) => Object.assign(p, { [k]: 0 }), {});
+    return dataResourcesLoaded && !dataResourcesError
+      ? dataResources?.reduce((drMap, dr) => {
+          const provider = dr?.spec?.type;
+          drMap[NOOBAA_PROVIDER_MAP[provider]] =
+            drMap[NOOBAA_PROVIDER_MAP[provider]] + 1;
+          return drMap;
+        }, initialCount)
+      : initialCount;
+  }, [dataResources, dataResourcesLoaded, dataResourcesError]);
```
Comment (cloudbehl) [inline packages/components/dashboard/resource-provider-card/resource-provider-card.tsx:40]: resource provider can be fetched from prometheus query. You will not require this logic

File: packages/components/dashboard/resource-providers-card/resource-providers-card.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+      <CardHeader>
+        <CardTitle>{t('Resource Providers')}</CardTitle>
+      </CardHeader>
+      <CardBody>
+        {isLoading ? (
+          <Skeleton />
+        ) : error || !hasProviders ? (
+          <div className="nb-resource-providers-card__not-available text-secondary">
+            {t('Not available')}
+          </div>
```
Comment (SanjalKatiyar) [inline packages/components/dashboard/resource-providers-card/resource-providers-card.tsx:99]: use ResourceProvidersBody component for this

File: locales/en/plugin__mcg-ms-console.json [hunk 3]
```diff
@@ -26,6 +26,10 @@
   "Select a single data source that defines the write targets of the bucket policy": "Select a single data source that defines the write targets of the bucket policy",
   "Data Federation": "Data Federation",
   "Overview": "Overview",
+  "Resources Providers": "Resources Providers",
```
Comment (SanjalKatiyar) [inline locales/en/plugin__mcg-ms-console.json:29]: where is this used ??
shouldn't this be `Resource Providers` ?

File: packages/components/dashboard/resource-providers-card/resource-providers-card.tsx [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+  providers.forEach((provider) => {
+    providersList[getProviderType(provider)] = getProviderCount(provider);
+  });
+  return providersList;
+};
+
+const ResourceProviders: React.FC<{}> = () => {
+  const { t } = useTranslation();
+
+  const [providersTypesQueryResult, providersTypesQueryResultError] =
```
Comment (SanjalKatiyar) [inline packages/components/dashboard/resource-providers-card/resource-providers-card.tsx:51]:
```suggestion
  const [providersTypesResult, providersTypesError, providersTypesLoading] =
```

File: packages/components/dashboard/resource-providers-card/resource-providers-card.tsx [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    });
+  const [
+    unhealthyProvidersTypesQueryResult,
+    unhealthyProvidersTypesQueryResultError,
+  ] = useCustomPrometheusPoll({
+    query: RESOURCE_PROVIDERS_QUERY.UNHEALTHY_PROVIDERS_TYPES,
+    endpoint: 'api/v1/query' as any,
+    basePath: MCG_MS_PROMETHEUS_URL,
+  });
+  const [resourcesLinksResponse, resourcesLinksResponseError] =
```
Comment (SanjalKatiyar) [inline packages/components/dashboard/resource-providers-card/resource-providers-card.tsx:65]:
```suggestion
  const [resourcesLinksResponse, resourcesLinksError, resourcesLinksLoading] =
```

File: packages/components/dashboard/resource-providers-card/resource-providers-card.tsx [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+  const allProviders = createProvidersList(providersTypesQueryResult);
+  const unhealthyProviders = createProvidersList(
+    unhealthyProvidersTypesQueryResult
+  );
+
+  const providerTypes = filterProviders(allProviders);
+  const hasProviders = !_.isEmpty(allProviders);
+  const isLoading =
+    !error &&
+    !(providersTypesQueryResult && unhealthyProvidersTypesQueryResult);
```
Comment (SanjalKatiyar) [inline packages/components/dashboard/resource-providers-card/resource-providers-card.tsx:87]:
```suggestion
  const isLoading = providersTypesLoading || unhealthyProvidersTypesLoading || resourcesLinksLoading;
```

File: packages/constants/prometheus.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import { WindowObject } from '../types';
+import { ONE_HOUR } from './common';
+
+export const MCG_MS_PROMETHEUS_URL =
+  '/api/proxy/plugin/mcg-ms-console/mcg-managed-prom';
```
Comment (SanjalKatiyar) [inline packages/constants/prometheus.ts:5]:
```suggestion
  '/api/proxy/plugin/mcg-ms-console/prometheus-proxy';
```
Backend is suggesting to use this now !!

File: packages/components/dashboard/resource-providers-card/resource-providers-card.tsx (hunk 2 above)
Comment (debjyoti-pandit) [reply packages/components/dashboard/resource-providers-card/resource-providers-card.tsx:100]: for `<div className="nb-resource-providers-card__not-available text-secondary">
            {t('Not available')}
          </div>`
this much code, I don't feel it is fruitful to use a new component

Comment (SanjalKatiyar) [reply packages/components/dashboard/resource-providers-card/resource-providers-card.tsx:100]: not blocking this PR.

**Pull request merged**